src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -217,6 +232,47 @@ public function catalogAction(
         );
     }
 
+    /**
+     * @param FormView $categoriesFormView
+     *
+     * @return array|null
+     */
+    private function extractSelectCategory(FormView $categoriesFormView)
```

Comment:
please extract private functions to dedicated services, the long story is to reduce the length of this messy controller, not to increase it!